AccessControl: Document new permissions restricting data source access. (#39091)
grafana/grafana: docs/sources/enterprise/access-control/fine-grained-access-control-references.md

| `fixed:settings:admin:read`     | `settings:read`                                                                                                                                                                                                                                                              | Grant access to read settings.                                                                                                                                                  |
| `fixed:settings:admin:edit`     | All permissions from `fixed:settings:admin:read` and<br>`settings:write`                                                                                                                                                                                                     | Grant same accesses as `fixed:settings:admin:read` and grants, in addition, access to update settings.                                                                          |

> | `fixed:datasources:editor:read` | `datasources:explore`                                                                                                                                                                                                                                                        | Grant access to explore data sources.                                                                                                                                           |

Author: ```suggestion
| `fixed:datasources:editor:read` | `datasources:explore`                                                                                                                                                                                                                                                        | Grant access to the **Explore** tab.                                                                                                                                           |
```

| Grafana Admin  | `fixed:permissions:admin:edit`<br>`fixed:permissions:admin:read`<br>`fixed:provisioning:admin`<br>`fixed:reporting:admin:edit`<br>`fixed:reporting:admin:read`<br>`fixed:users:admin:edit`<br>`fixed:users:admin:read`<br>`fixed:users:org:edit`<br>`fixed:users:org:read`<br>`fixed:ldap:admin:edit`<br>`fixed:ldap:admin:read`<br>`fixed:server:admin:read`<br>`fixed:settings:admin:read`<br>`fixed:settings:admin:edit` | Default [Grafana Server Admin]({{< relref "../../permissions/_index.md#grafana-server-admin-role" >}}) assignments. |
| Admin          | `fixed:users:org:edit`<br>`fixed:users:org:read`<br>`fixed:reporting:admin:edit`<br>`fixed:reporting:admin:read`<br>`fixed:datasources:admin`                                                                                                                                                                                                                                                                               | Default [Admin]({{< relref "../../permissions/organization_roles.md" >}}) assignments.                              |

> | Editor         | `fixed:datasources:editor:read`                                                                                                                                                                                                                                                                                                                                                                                             | Default [Editor]({{< relref "../../permissions/organization_roles.md" >}}) assignements.                            |

Reviewer: Fix all instances of the typo "assignements" please. :)

| `fixed:datasource:editor:read`        | `datasources:explore`                                                                                                                                                                                                                                                        | Allows to access the **Explore** tab                                                                                                      |
| `fixed:datasources:admin`             | `datasources:read`<br>`datasources:create`<br>`datasources:write`<br>`datasources:delete`                                                                                                                                                                                    | Allows to create, read, update, delete data sources.                                                                                      |

> | `fixed:datasources:id:viewer`         | `datasources:id:read`                                                                                                                                                                                                                                                        | Allows to read data sources ID.                                                                                                           |

Reviewer 2: Minor: I think `data source IDs` sounds slightly better?

Author: ```suggestion
| `fixed:datasources:id:viewer`         | `datasources:id:read`                                                                                                                                                                                                                                                        | Allows to read data source IDs.                                                                                                           |
```

Author: Could it be `data sources' IDs`? Because each data source has 1 ID?

Author: ```suggestion
| `fixed:datasources:id:viewer`         | `datasources:id:read`                                                                                                                                                                                                                                                        | Allows to read data source IDs.                                                                                                           |
```

> | `fixed:datasources:permissions:admin` | `datasources.permissions:create`<br> `datasources.permissions:read`<br> `datasources.permissions:delete`<br>`datasources.permissions:toggle`                                                                                                                                 | Allows to create, read, delete, enable, or disable data source permissions                                                                |

Author: Modified these

Reviewer 3: I like that we are very explicit about what `fixed:datasource:editor:read` role allows to do. I assume in the future it might have more permissions assigned, however good to be clear about it no.

## Default built-in role assignments

> | Built-in role | Associated role                                                                                                                                                                                                                                                                                                                                                                                                             | Description                                                                                                                 |

Author: Modified descriptions and added two roles to admin `fixed:datasources:admin` `fixed:datasources:permissions:admin`
one role to Viewer: `fixed:datasources:id:viewer`
